src/inmanta/env.py

```diff
@@ -696,6 +696,12 @@ def init_env(self) -> None:
                 raise VenvCreationFailedError(msg=f"Unable to create new virtualenv at {self.env_path}")
             LOGGER.debug("Created a new virtualenv at %s", self.env_path)
         elif not os.path.exists(self._path_pth_file):
+            if not os.path.exists(self.site_packages_dir):
+                # the venv hosts a different python version than the running process
+                raise VenvActivationFailedError(
+                    msg=f"Unable to use virtualenv at {self.env_path} due to python version mismatch"
+                )
```

Comment:
This check only works for Linux operating systems, because the python version is present in the path to the `site-packages` directory. On windows this check would not work (See: https://github.com/inmanta/inmanta-core/blob/6316212832ee0c295cc98a055f35944d10908268/src/inmanta/env.py#L225), but I didn't check this on a Windows machine. I think we should implement this check in a platform independent way.

Reply:
After a discussion with @arnaudsjs we came up with these 2 solutions to check out the python version on a Windows machine:
- Parsing the `pyenv.cfg` file (very fragile solution because no guarantees on the format of the file)
- Forking a new process and executing `<path-to-python-binary> --version` (Slow operation and potential impact on the runtime of the test cases)
@wouterdb  @sanderr what do you think ?

Reply:
How about checking for the fully qualified python binary? I'm not 100% sure but I think it should always be present (e.g. in a Python 3.9 venv there should always be a `bin/python3.9`.

Reply:
Ok so on windows it seems the equivalent of the `ENV/bin` folder is the `ENV\Scripts` folder.
Here is its content:
```bash
Directory of C:\Users\hugo\tmp\test_env_310\Scripts

02/03/2022  10:08    <DIR>          .
02/03/2022  10:08    <DIR>          ..
02/03/2022  10:08             2.086 activate
02/03/2022  10:08             1.015 activate.bat
02/03/2022  10:08            20.571 Activate.ps1
02/03/2022  10:08               393 deactivate.bat
02/03/2022  10:08           106.360 pip.exe
02/03/2022  10:08           106.360 pip3.10.exe
02/03/2022  10:08           106.360 pip3.exe
02/03/2022  10:08           263.056 python.exe
02/03/2022  10:08           251.792 pythonw.exe
               9 File(s)        857.993 bytes
               2 Dir(s)   8.019.701.760 bytes free
```
               
Do you think it's ok/safe to use `pip3.10.exe` to infer the version of python ?

There are also lots of files in `\Lib\site-packages\pip\*\__pycache__` containing the python version in their names, but then again I don't know how robust it is to use them as such.

For example `__init__.cpython-310.pyc` in `C:\Users\hugo\tmp\test_env_310\Lib\site-packages\pip\_internal\cli\__pycache__`

Reply:
Ah, shame. I don't think that would be a safe check because you can create a venv without pip (and I think the compiler might even actually do so). So back to `python --version` I guess. I just had a chat with @wouterdb and he proposed to try the performant check (the one in your current implementation) and if that fails fall back to the `--version` one. This way the tests wouldn't suffer either.